generator/src/main/java/com/scottlogic/deg/generator/generation/DecisionTreeDataGenerator.java

```diff
@@ -56,4 +67,32 @@ public DecisionTreeDataGenerator(
             .limit(maxRows)
             .peek(monitor::rowEmitted);
     }
+
+    private DecisionTree reportThenCullContradictions(DecisionTree decisionTree) {
+        Collection<Node> contradictingNodes = treeValidator.reportContradictions(decisionTree);
+        if (contradictingNodes.contains(decisionTree.getRootNode())) {
+            // Entire profile is contradictory.
+            monitor.addLineToPrintAtEndOfGeneration(
+            "The provided profile is wholly contradictory. No fields can successfully be fixed.",
+                System.err
+            );
+            return new DecisionTree(null, decisionTree.getFields(), decisionTree.getDescription());
+        } else if (contradictingNodes.size() > 0) {
+            // Part of the profile is contradictory, and therefore could be simplified.
+            monitor.addLineToPrintAtEndOfGeneration(
+            "Warning: There are " +
+                contradictingNodes.size() +
+                " partial contradiction(s) in the profile. The contradicting section(s) are:",
+                System.err
+            );
+            for (Node node : contradictingNodes) {
+                monitor.addLineToPrintAtEndOfGeneration(node.toString(), System.err);
+            }
+            // TODO: Remove all contradicting subtrees in the decision tree and return the new tree.
```

Comment:
this can be done by calling 
reductiveTreePruner.pruneConstraintNode(rootnode, [a map of each contradicting field -> empty fieldspec]